### PR TITLE
Correct intermittent fault in TLS test case

### PR DESCRIPTION
**Pull request by bjosv**
Depending on timing the connection state in testcase t_tls_1_3_cert_expired
is either to-be-closed or already closed.

### CODE REVIEW

File: test/eredis_tls_SUITE.erl [hunk 1]
```diff
@@ -96,9 +96,15 @@ t_tls_1_2_cert_expired(Config) when is_list(Config) ->
...
-    %%io:format(user, "## ~p~n", [ssl:cipher_suites(all, 'tlsv1.3')]),
     C = c_tls(ExtraOptions, CertDir, [{versions, ['tlsv1.3']}]),
-    ?assertMatch({error, {tls_alert, {certificate_expired, _}}}, eredis:q(C, ["GET", foo])),
+    %% Depending on timing the query will result in a tls_alert error,
+    %% but if the error indication was handled before sending the query
+    %% the connection has already been taken down, hence a no_connection
+    case eredis:q(C, ["GET", foo]) of
+        {error, {tls_alert, {certificate_expired, _}}} -> ok;
+        {error, no_connection} -> ok;
+        Res -> ?assert(false, Res)
```
Comment (zuiderkwast) [inline test/eredis_tls_SUITE.erl:106]:
```suggestion
        Res -> ?assertMatch({error, no_connection}, Res)
```

**Review by zuiderkwast — APPROVED** [review]
Nice!

Isn't there a way to make this deterministic?

**Comment by bjosv** [comment]
> Nice!
> 
> Isn't there a way to make this deterministic?

A sleep :)

**Comment by zuiderkwast** [comment]
> > Nice!
> > Isn't there a way to make this deterministic?
> 
> A sleep :)

Add a sleep then? :sleeping: In the makefile or in the test suite? `timer:sleep(Milliseconds)`.

**Comment by bjosv** [comment]
The error will arrive in the `handle_info\3` for ssl_error, and this will trigger a disconnection. If there are queries outstanding these will be aborted using `reply_all` with `{error, Reason}` as code. If the query is a bit late it will get no_connection directly instead

**Comment by zuiderkwast** [comment]
> The error will arrive in the `handle_info\3` for ssl_error, and this will trigger a disconnection. If there are queries outstanding these will be aborted using `reply_all` with `{error, Reason}` as code. If the query is a bit late it will get no_connection directly instead

I understand. Let's keep it as it is now then. Let's merge! :100:

**Comment by bjosv** [comment]
> > > Nice!
> > > Isn't there a way to make this deterministic?
> > 
> > 
> > A sleep :)
> 
> Add a sleep then?  In the makefile or in the test suite? `timer:sleep(Milliseconds)`.

The sleep would be needed in the testcase, right before the query is triggered but after a connection has been setup. The testcase was created to test a query direct after the connect due to the different behaviour in TLS 1.3 compared to TLS 1.2. In 1.2 we get the error during the connect setup directly, and this testcase would expect the error to arrive just after the connection. Running locally this always triggered as in the testcase before, but I guess CI is a bit slower.

**Pull request merged**